### DISCUSSION TITLE
Legend should be available offline (for instance in details pointing to the object)

### DISCUSSION THREAD

**Discussion by starbrights** [osmandapp/OsmAnd#12959] [Ideas]
Thanks for this great tool, first of all. I am using it quite a while and still discover new features.
I would ask if it would be possible to include a legend (matching the selected kind of display (Mapnik, Topo ...).
That should be available offline. Either as a document of even better if you point to it in details. At the moment I get there just the position. I think that would be a cool feature help to work with OSMand.
As for me I am usually getting confused about consistence of ways and some rare uses icons.
Thanks again, and maybe one of the  devs can give feedback whether that would be a good idea or is impossible or solved in any other way I didn't found until now.

**Comment by mariush444** [comment]
forget ... there is long term issue about that. Starting from 2015.
Do not expect to much.

During discussion with one user it was even something like a promise on Telegram chat

![image](https://user-images.githubusercontent.com/66887280/137471056-7a9ba0ab-659d-4506-9f70-90facac05886.png)

additionally maplegend in osmand is not equal the the rendering in osmand.
https://github.com/osmandapp/OsmAnd/discussions/12525

**Reply by EugeneZmeuk** [reply]
Hm, 
Yes,  Legend for offline use is our goal. 
Now, we have something interesting, it's the custom plugin with map legend. 
You can write to me (EZ) and I will send you this plugin for testing.

**Reply by starbrights** [reply]
Hey EZ, thanks for offering. I would give it a try, but couldn't found a way to email you. Can you upload it somewhere? Or email me to star@gmx.li Thanks

**Reply by EugeneZmeuk** [reply]
Done

**Reply by starbrights** [reply]
Works, thank you!

**Reply by starbrights** [reply]
Sorry to ask - I'm somewhat lost. I remeber I need some time to find the Legend, but I don't find it again. Where to I have to look for it?
Is this Legend just for OSMand map style? At the moment I go for Offroad style as the little ways keep visible if zoomed out.

**Reply by mariush444** [reply]
Location: geo:45.738777,36.493324?z=17
https://osmand.net/go?lat=45.738777&lon=36.493324&z=17

**Comment by mariush444** [comment]
@EugeneZmeuk 
Thank you for plugin.

What is the relation between map legend and rendering in osmand?

There is openstreetmap tag for the railway - monorail
![image](https://user-images.githubusercontent.com/66887280/137793807-e323e66c-4d8d-494d-93f9-09140b9e08d8.png)

This is what I see in Osmand
![image](https://user-images.githubusercontent.com/66887280/137794033-aed65d37-a376-4f7c-a0a3-1c14f6591318.png)

and the legend definition
![image](https://user-images.githubusercontent.com/66887280/137794291-f1627102-8518-476a-867f-60548b5cf6b4.png)

PS
Can user move legend to other place?
Then I can show osm element and the legend in the same moment/picture

**Reply by EugeneZmeuk** [reply]
OsmAnd has its own rendering engine. Map legend is OsmAnd rendering.
You can change Map style in OsmAnd = rendering style, of course. (You see how it will be changed for Map Legend elements because Plugin Map legend is a vector.) 
- You can not move The legend to other places without special skills (JOSM or OsmAnd Map creator).
- hm, maybe you can use OSM tiles (online map layer - overlay/underlay)

**Reply by EugeneZmeuk** [reply]
I hope I answered your questions.

**Reply by karl-01** [reply]
How can I get this plugin? 
karl_krause@gmx.net

Thanks in advance
Karl

**Comment by mariush444** [comment]
@EugeneZmeuk 
Are we allowed to share this plugin?

**Reply by EugeneZmeuk** [reply]
of course

**Comment by mariush444** [comment]
[legend.zip](https://github.com/osmandapp/OsmAnd/files/7403144/legend.zip)
after installation the plugin,  it needs to download in `Download maps menu -> legend`
the legend can be seen at
**Location: geo:45.738777,36.493324?z=17**
https://osmand.net/go?lat=45.738777&lon=36.493324&z=17

_advice: make a favorite at this place to easier find the legend next time_

**Reply by EugeneZmeuk** [reply]
1. Unzip Legend.zip -> Map legend plugin.osf

**Comment by karl-01** [comment]
Hi Mars Yush,

 

thanks for the legend-file and the advice for Download.

But currently I do not know, how to proceed.

 

Can you please give me some hint, how to install this plugin (Map Legend Plugin.osf)?

Up to now I have only enabled/disabled built-in plugins.

 

The following steps, I hope, will then become more clear to me.

 

Thanks in advance

Karl

 

Von: Mars Yush ***@***.*** 
Gesendet: Samstag, 23. Oktober 2021 16:28
An: osmandapp/OsmAnd
Cc: karl-01; Comment
Betreff: Re: [osmandapp/OsmAnd] Legend should be available offline (for instance in details pointing to the object) (Discussion #12959)

 

legend.zip <https://github.com/osmandapp/OsmAnd/files/7403144/legend.zip> 
after installation the plugin, it needs to download in Download maps menu -> legend
the legend can be seen at
Location: geo:45.738777,36.493324?z=17
https://osmand.net/go?lat=45.738777 <https://osmand.net/go?lat=45.738777&lon=36.493324&z=17> &lon=36.493324&z=17

advice: make a favorite at this place to easier find the legend next time

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub <https://github.com/osmandapp/OsmAnd/discussions/12959#discussioncomment-1525209> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AC2XV6YVZ2UCUNVLDZBGMRDUILBAXANCNFSM5FT3ITUA> .
Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>  or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub> . Das Bild wurde vom Absender entfernt.

**Reply by mariush444** [reply]
unzip the file, then just click/start the osf file to open the file in osmand
osmand will install the plugin

**Reply by starbrights** [reply]
I think as long as you have to remember the location to see the legend it is not useful? Why this is not official part, or at least always accessable?

**Reply by EugeneZmeuk** [reply]
You need to add Favorite point near Legend.
While we think about how to use Map legend for offline use. This variant with Plugin is a test version for us.  You need to understand that The legend must be Vector style. I hope we will add The Legend in the first part of 2022 by default.

**Reply by buffoletti** [reply]
Can anybody point me in the right direction with this plugin: It seems to be properly installed and active, but nothing is visible in the location.
What does this mean, do I have to create a folder somewhere?

> it needs to download in Download maps menu -> legend

**Comment by fabeit** [comment]
@mariush444 Can any of the devs answer why it's not possible to add a few Kbs of image that could be used for a map legend? What's the need of having to download the legend every time? Why can it not be cached? It's puzzling.

**Reply by vshcherb** [reply]
Please anybody who knows how to fix it, please feel free to send PR.

**Reply by rlaggren** [reply]
I suspect the download is a test, a working prototype that's good enough to put out there for use by those who have the interest.

It has not been implemented into the main program features yet, but it's on near-future plan. Those Who Know correct  me as needed...


Rufus

**Reply by mariush444** [reply]
@vshcherb I see that current plugin installs legeng.obf so it is solution for offline map as I understand.
But if anyone needs image then idea of solution is in this file
[map-legend-roads.gpx.zip](https://github.com/osmandapp/OsmAnd/files/7576545/map-legend-roads.gpx.zip)
This is embedded image in html. Html code is included in gpx file that is based on your travel guide idea.
The problem is that it is not "a few Kbs of image"
Osmand crashes when big image is used.

The second possibility is to put map legend image to avnotes folder
[map-legend-poi.zip](https://github.com/osmandapp/OsmAnd/files/7576639/map-legend-poi.zip)
then the picture can be big enough.

**For me, the best solution is to have possibility to create gpx file with included html code but 
html shouldn't be with embedded (base64) image but with path to the file in osmand/avnotes folder.** 

Currently path to picture can not be used in html code embedded into gpx file.

**Reply by vshcherb** [reply]
Yes big images are the issue for htmls - please use SVG! It's small enough and should be rendered. We had that idea but we didn't have time to create it... And we've focused on the Map rendered in the black sea as an example

**Reply by fabeit** [reply]
The imagine can be a few MB, definitely nothing that breaks a smartphone with many Gb of storage, this is just nonsense!

**Comment by karl-01** [comment]
Thank you, Eugene,



I will try..



Karl



Von: Eugene Kizevich ***@***.***
Gesendet: Montag, 25. Oktober 2021 11:30
An: osmandapp/OsmAnd
Cc: karl-01; Comment
Betreff: Re: [osmandapp/OsmAnd] Legend should be available offline (for
instance in details pointing to the object) (Discussion #12959)



You need to add Favorite point near Legend.
While we think about how to use Map legend for offline use. This variant
with Plugin is a test version for us. You need to understand that The legend
must be Vector style. I hope we will add The Legend in the first part of
2022 by default.

-
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/osmandapp/OsmAnd/discussions/12959#discussioncomment-153
0919> , or unsubscribe
<https://github.com/notifications/unsubscribe-auth/AC2XV6222BTHPUEYLAD6VUTUI
UPPTANCNFSM5FT3ITUA> .
Triage notifications on the go with GitHub Mobile for iOS
<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&m
t=8&pt=524675>  or Android
<https://play.google.com/store/apps/details?id=com.github.android&referrer=u
tm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>
. Das Bild wurde vom Absender entfernt.

**Comment by Max1234-Ita** [comment]
I tried the plugin, it's a very good idea, even though the produced symbols are very small in size; They should be bigger (or size should be adjustable) for better readability.

Just for information: is there any equivalent of this plugin working on OsmAnd for iOS?

Thanks a lot for your work,
Max

**Comment by fabeit** [comment]
I don't understand how this works, do you have to preregister the legend only at specific points?

**Reply by Max1234-Ita** [reply]
> I don't understand how this works, do you have to preregister the legend only at specific points?

Just point your map to this position: https://osmand.net/go?lat=45.738777&lon=36.493324&z=17

It's somewhere in the middle of the Sea of Azov. That's why you need to place a favorite to this position (it's not very comfortable to check up the legend and go back to your previous location, so any native solution would be appreciated 🙂 )

**Reply by buffoletti** [reply]
Doesn't seem to work for me. Plugin is installed and active, but nothing but blue sea at the given location...😕

**Reply by buffoletti** [reply]
Here is what I did:
1. Download and extract the "Map Legend Plugin.osf" 
3. Open with osmand (straight from downloads folder, not moving the file anywhere eg. in the app folder)
4. Verify plugin is installed and active (yes)
5. Go to location in Azov sea
6. No legend

**Reply by Max1234-Ita** [reply]
> Here is what I did:
> 
>     1. Download and extract the "Map Legend Plugin.osf"
> 
>     2. Open with osmand (straight from downloads folder, not moving the file anywhere eg. in the app folder)
> 
>     3. Verify plugin is installed and active (yes)
> 
>     4. Go to location in Azov sea
> 
>     5. No legend

The plugin also installs a new "map" which actually contains the Legend. 
Try checking if it is enabled in _/// -> Maps and resources menu -> Local_ (_-> Standard maps_, if I remember well)

**Reply by buffoletti** [reply]
@Max1234-Ita 
That's it! Thanks a lot! Ist a New map, that also needs to be downloaded from within the app manually. Maybe because I wasnt on wifi, I needed to confirm the download...

**Reply by Max1234-Ita** [reply]
> @Max1234-Ita That's it! Thanks a lot! Ist a New map, that also needs to be downloaded from within the app manually. Maybe because I wasnt on wifi, I needed to confirm the download...

Glad it helped 🙂
